FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php

```diff
@@ -225,4 +225,53 @@ public function testHydrateFacet(
             );
         }
     }
+
+    /**
+     * Data provider for testing removal of control characters
+     *
+     * @return array
+     */
+    public static function controlCharacterProvider()
```

Comment:
Seems that the story description was wrong (regarding that this provider can be copied). Use this instead:
```
    public static function unescapedFilterNameProvider()
    {
        return [
            'Filter name with only letters and numbers' => [
                'Findologic123',
                'Findologic123',
                'Expected string to return unchanged'
            ],
            'Filter name with spaces' => [
                "Findologic 1 2 3",
                'Findologic_1_2_3',
                'Expected whitespaces to be stripped way'
            ],
            'Filter name with dots' => [
                "Findologic...Rocks",
                'Findologic_Rocks',
                'Expected dots to be stripped way'
            ],
            'Filter name with special characters' => [
                'Findologic&123',
                'Findologic&123',
                'Expected special characters to be returned as they are'
            ],
            'Filter name with non standard character' => [
                'Findologic@123',
                'Findologic_123',
                'Expected non standard characters to be stripped away'
            ],
            'Filter name with umlauts' => [
                'Findolögic123',
                'Findolögic123',
                'Expected umlauts to be left unaltered.'
            ]
        ];
    }
```